### PR TITLE
RFC 5424 Logging Support

### PR DESCRIPTION
**Pull request by stylewarning**
This PR is adding [RFC 5424](https://www.rfc-editor.org/rfc/pdfrfc/rfc5424.txt.pdf) compliance to CL-SYSLOG. It does so by building atop existing functionality. In particular, it adds a new `rfc5424-logger` class, along with a `log-string` function and an `rfc-log` macro.

It is *incomplete* and ~*not ready to merge*~. There are still some TODO items:

- [x] Clean up the file a bit.
- [x] Figure out where and how streams should be used
- [x] Add README documentation
- [ ] Add tests

With that said, all features of RFC5424 are present, including structured data, and log processors seem to understand it OK.

### CODE REVIEW

**Comment by mmaul** [comment]
Wow, love your code, comments very well written!!!
So thoughts:
I like your interface, especially the support for structured data

Since you switched from dot to slash as the package and "sub-package" separator it would be good to idea maintain the legacy package names for a while, but indicate they are depreciated. Not that it matters that much I don't think there is anything in quicklisp that relies on cl-syslog (I checked quickdocs, as a quick way to look). Frankly I like your slashes better.....

Another TODO might also be to implement TCP syslog if you want to go all in implement RFC 5425 Transport Layer Security (TLS) Transport Mapping for Syslog

If your goal is high performance computing you might consider a buffering mechanism to limit the number of network requests or syslog calls. Going a littler further decouple things a bit and make client interface coupled via message passing to the logging system and run that in a separate thread. Course this may be a little bit of overkill unless your goal is to support high throughput logging.

Perhaps set a default value for facility in rfc5424-logger, I'm thinking user

What might be nice is create a cl-syslog appender for log4cl.

Finally it seems like you have vision and perhaps a vested interest in this package so if you want to take over this project I could transfer the project over to you if you are interested.

**Review by mmaul — COMMENTED** [review]
Lke slashes better but should also keep systems with dots to keep from breaking code that may depend on cl-syslog, however should not be much.

**Review by mmaul — APPROVED** [review]
Lke slashes better but should also keep systems with dots to keep from breaking code that may depend on cl-syslog, however should not be much.

**Comment by stylewarning** [comment]
Note that the slashes are for ASDF conformance. I agree, it wouldn't hurt to make backward compat system names (in separate files) which can be deprecated and later deleted.

**Comment by stylewarning** [comment]
I am going to hold off adding tests. I don't want to figure out `NST` yet, and I'd probably prefer to replace it wholesale with a much simpler testing framework like Fiasco.

**Comment by stylewarning** [comment]
@mmaul I have implemented some of your requests:

1. I made the default facility `:USER`.
2. I created backwards compatible ASDF packages with the dotted-names. Note that the slashes are now ASDF-standard.
3. I decided not to touch the UDP package at all. It does cause compile-time style warnings.
4. I decoupled the log generation from the log transmission. Check out the `log-writer` slot, which can take any function at all. By default, it just calls into the normal `syslog`.
5. I wrote a bit of a longer README.

I think this is ready for merging, if you're OK with it not having extensive tests. I'm OK to be on the hook for fixing bugs, and slowly adding tests as time goes on.

EDIT: I should mention that after approval, I'm happy to squash commits to make a cleaner commit history.

File: README.md
Comment (notmgsk) [inline README.md]: trailing `

File: README.md
Comment (notmgsk) [inline README.md:27]: "on" or "or"? Might be correct as is

File: README.md
Comment (notmgsk) [inline README.md]: That is not 5 digits. Maybe say "by a usually (but not necessarily) 5 digit integer"

File: README.md
Comment (notmgsk) [inline README.md]: "It also expects that your..."

**Review by notmgsk — CHANGES_REQUESTED** [review]

File: rfc5424-reserved.lisp
Comment (notmgsk) [inline rfc5424-reserved.lisp]: Do it

File: README.md
Comment (stylewarning) [reply README.md:27]: on is correct

**Review by notmgsk — APPROVED** [review]

File: README.md
Comment (notmgsk) [inline README.md:164]: expected

**Pull request merged**
